Supertest: getTestServer Not Found

Hello and to whom it may concern.

I am trying to work with the implementation of [getTestServer in Supertest documentation](https://github.com/GoogleCloudPlatform/functions-framework-nodejs/blob/main/docs/testing-functions.md#integration-testing-with-supertest), and when I try to instantiate my test server I get this error:


<img width="1556" alt="Screenshot 2024-03-25 at 9 58 27 AM" src="https://github.com/GoogleCloudPlatform/functions-framework-nodejs/assets/1054644/fe0f1141-1258-46d1-942c-cc485ce92d33">

@google-cloud/functions-framework is installed as a dependency of this project so my assumption would have been that this should be available in the import.

I've Googled a few things and I've tried updating my tsconfig, but nothing seems to work. 

Thanks ahead.

When I did that the terminal threw an error saying: "error TS5110: Option 'module' must be set to 'NodeNext' when option 'moduleResolution' is set to 'NodeNext'." I did that and got it to compile correctly, but then my test threw a jest configuration error"

 FAIL  __tests__/endpoints.spec.ts
  ● Test suite failed to run

    Unable to process '/Users/adampurdy/Sites/KPA_PRODUCTIONS/USSSA/ms-platform/event-api-ms/__tests__/endpoints.spec.ts', please make sure that `outDir` in your tsconfig is neither `''` or `'.'`. You can also configure Jest config option `transformIgnorePatterns` to inform `ts-jest` to transform /Users/blah/Sites/blah/blah/ms-platform/event-api-ms/__tests__/endpoints.spec.ts

      at TsCompiler.getCompiledOutput (node_modules/ts-jest/dist/legacy/compiler/ts-compiler.js:173:27)
      at TsJestCompiler.getCompiledOutput (node_modules/ts-jest/dist/legacy/compiler/ts-jest-compiler.js:14:39)
      at TsJestTransformer.exports.TsJestTransformer.TsJestTransformer.processWithTs (node_modules/ts-jest/dist/legacy/ts-jest-transformer.js:265:37)
      at TsJestTransformer.exports.TsJestTransformer.TsJestTransformer.process (node_modules/ts-jest/dist/legacy/ts-jest-transformer.js:188:24)
      at ScriptTransformer.transformSource (node_modules/@jest/transform/build/ScriptTransformer.js:545:31)
      at ScriptTransformer._transformAndBuildScript (node_modules/@jest/transform/build/ScriptTransformer.js:674:40)
      at ScriptTransformer.transform (node_modules/@jest/transform/build/ScriptTransformer.js:726:19)

I tried adjusting jest.config.js and that didn't work either.

Can you help me understand why I need to adjust the tsconfig, when I'm importing other @google-cloud/functions-framework functions like:

import { cloudEvent } from '@google-cloud/functions-framework'. That works fine without adjusting my tsconfig from what it was.

Thanks

--- a/package.json
+++ b/package.json
@@ -18,6 +18,13 @@
       "default": "./build/src/testing.js"
     }
   },
+  "typesVersions": {
+    "*": {
+      "testing": [
+        "./build/src/testing.d.ts"
+      ]
+    }
+  },
   "dependencies": {
     "@types/express": "^5.0.0",
     "body-parser": "^2.2.2",

